relayer/src/services/relayer/optimism.ts

```diff
@@ -75,6 +90,90 @@ export class OptimismRelayerService extends BaseRelayerService {
     return;
   }
 
+  private async getMessageWithdrawalHash(message: CrossChainMessage): Promise<string> {
+    const crossChainMessage = await this.messenger.toBedrockCrossChainMessage(message);
+    const lowLevelMessage = await this.messenger.toLowLevelMessage(crossChainMessage);
+
+    return hashLowLevelMessage(lowLevelMessage);
+  }
+
+  private async isMessageWithdrawn(messageHash: string) {
+    const OptimismPortal = this.messenger.contracts.l1.OptimismPortal.connect(this.l1Wallet);
+    const isWithdrawn = await OptimismPortal.finalizedWithdrawals(messageHash);
+
+    return isWithdrawn;
+  }
+
+  private async relayMessageViaCrossDomainMessenger(
```

Comment:
I would add some docblocks explaining the quirks of the optimism contracts. If it wasn't for @fredo TLDR-ing me on the problem I wouldn't understand why we need to use different contracts for basically the same thing. 
3 months from now we wouldn't know it either. 

Other than that - it's great!